src/main/scala/cromwell/engine/workflow/ValidateActor.scala

```diff
@@ -0,0 +1,55 @@
+package cromwell.engine.workflow
+
+import akka.actor.{Actor, Props}
+import com.typesafe.scalalogging.LazyLogging
+import cromwell.binding._
+import cromwell.webservice.PerRequest.RequestComplete
+import cromwell.webservice.{WorkflowJsonSupport, WorkflowValidateResponse}
+import spray.http.StatusCodes
+import spray.httpx.SprayJsonSupport._
+import spray.json._
+
+import scala.concurrent.Future
+import scala.language.postfixOps
+import scala.util.{Failure, Success}
+
+object ValidateActor {
+  sealed trait ValidateActorMessage
+  case object ValidateWorkflow extends ValidateActorMessage
+
+  def props(wdlSource: WdlSource, wdlJson: WdlJson): Props = {
+    Props(new ValidateActor(wdlSource, wdlJson))
+  }
+}
+
+class ValidateActor(wdlSource: WdlSource, wdlJson: WdlJson)
+  extends Actor with LazyLogging {
+
+  import WorkflowJsonSupport._
+  import context.dispatcher
+  import ValidateActor.ValidateWorkflow
+  private val tag = "ValidateActor"
+
+  override def receive = {
+    case ValidateWorkflow =>
+      val sentBy = sender()
+      logger.info(s"$tag for $sentBy")
+      val futureInputs = for {
+        eventualNamespace <- Future(NamespaceWithWorkflow.load(wdlSource, WorkflowManagerActor.BackendType))
+        inputs <- Future(wdlJson.parseJson) map {_.asJsObject.fields}
+        coercedInputs <- Future.fromTry(eventualNamespace.coerceRawInputs(inputs))
+        _ <- Future.fromTry(eventualNamespace.staticDeclarationsRecursive(coercedInputs))
+      } yield ()
+
+      futureInputs onComplete {
+        case Success(_) =>
+          logger.info(s"$tag success $sentBy")
+          sentBy ! RequestComplete(StatusCodes.OK, WorkflowValidateResponse(valid = true, error = None))
+
+        case Failure(ex) =>
+          val messageOrBlank = Option(ex.getMessage).mkString
+          logger.info(s"$tag error $sentBy: $messageOrBlank")
+          sentBy ! RequestComplete(StatusCodes.BadRequest, WorkflowValidateResponse(valid = false, error = Option(messageOrBlank)))
```

Comment:
I played a bit with replacing this with `val messageOrBlank = if (ex.getMessage == null) "" else ex.getMessage`, but the current line was more compact, and didn't use the "evil" `null`.

As to why when `ex.getMessage == null` the error is reported as with `Some("")` instead of `None`, I preferred letting the caller know that there _was_ an exception, but that the message wasn't there.

We could instead return a json `"error": null` instead of `"error": ""`, or completely eliminate the key altogether like we do when return `"valid": true`, but my preference was:

```json
{
  "valid": false,
  "error": ""
}
```

This also matches the behavior in [`WorkflowManagerActor`](https://github.com/broadinstitute/cromwell/blob/f347e80b715b0959c52f5ce0abda09c549058a72/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala#L184)